plugins/i3s/io/I3SReader.cpp

```diff
@@ -90,9 +90,17 @@ void I3SReader::buildNodeList(std::vector<int>& nodes, int pageIndex)
     {
         BOX3D nodeBox = parseBox(nodeIndexJson["nodes"][i]);
         int cCount = nodeIndexJson["nodes"][i]["childCount"].asInt();
+        //density calculated as (number of points in node) / (lod threshold)
+        int pCount =  nodeIndexJson["nodes"][i]["vertexCount"].asInt();
+        double lodThreshold =
+            nodeIndexJson["nodes"][i]["lodThreshold"].asDouble();
+        double density = (double)pCount / lodThreshold;
         bool overlap = m_bounds.overlaps(nodeBox);
-        if (cCount == 0 && overlap)
+        if (density > m_args.lod && density < m_args.lod + 0.5 && overlap)
         {
+            std::cout << "lod: " << lodThreshold << std::endl;
```

Comment:
* Remove debugging lint
* Complex single-line conditional with no parens
* what does this magic `0.5` number do?

Reply:
0.5 doesn't really mean anything. I just picked an arbitrary threshold to limit it. 0.5 seemed like a good cutoff. Could have them enter a range for the lod.

Reply:
Otherwise you're getting a lot of overlapping resolutions if you pick greater than or less than one single number.